### PR TITLE
Feat/td 31877 ws interface update

### PR DESCRIPTION
**Pull request by YamingPei**
The external interface of taosws was upgraded and the ODBC was modified simultaneously

### CODE REVIEW

File: inc/taosws_helpers.h [hunk 1]
```diff
@@ -49,7 +49,7 @@
   WS_RES *__res  = res;                                              \
   int _e         = ws_errno(__res);                                  \
   const char *_s = ws_errstr(__res);                                 \
-  if (!_s || !*_s) break;                                            \
+  if (!_e) break;                                            \
```
Comment (freemine) [inline inc/taosws_helpers.h:52]: this was intentional at the very beginning, no matter whether _e is zero or not, we'll dump ws_errstr(__res) if it returns non-empty string.

File: inc/taosws_helpers.h [hunk 2]
```diff
@@ -58,7 +58,7 @@
   int _e          = ws_errno(NULL);                                                       \
   const char *_s1 = ws_errstr(NULL);                                                      \
   const char *_s2 = ws_stmt_errstr(__stmt);                                               \
-  if (!_s2 || !*_s2) break;                                                               \
+  if (!_e) break;                                                               \
```
Comment (freemine) [inline inc/taosws_helpers.h:61]: as above

File: inc/taosws_helpers.h (hunk 1 above)
Comment (YamingPei) [reply inc/taosws_helpers.h:52]: Okay, I'll restore to the original use,

**Comment by freemine** [comment]
LGTM

**Pull request merged**